### PR TITLE
250 ember mirage

### PR DESCRIPTION
**Pull request by allthesignals**
This PR adds and implements ember Mirage for offline testing and development. It also adds an acceptance test for project creation. 

All of the standard JSON:API resource endpoints are handled with Mirage. The Carto SQL requests are intercepted with mirage, and handled by searching a HAR file and returning simple matches.

### CODE REVIEW

File: frontend/lib/carto-promises-utility/addon/utils/carto.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import fetch from 'fetch';
+import ENV from 'labs-ceqr/config/environment';
 
-const cartoUsername = 'planninglabs';
-const cartoDomain = `${cartoUsername}.carto.com`;
-
+const cartoUsername = ENV.carto['username'];
+const cartoDomain = ENV.carto['domain'];
```
Comment (pichot) [inline frontend/lib/carto-promises-utility/addon/utils/carto.js]: <3 thank you

File: frontend/app/components/project-details.js [hunk 2]
```diff
@@ -6,7 +6,11 @@ export default Component.extend({
 
   buildYearRange: Array.from({length: (2040 - 2018)}, (v, k) => k + 2018),
 
+  // noop
+  save() { return; },
```
Comment (pichot) [inline frontend/app/components/project-details.js]: I know this is necessary, I just don't know why. Can you explain it to me?

Comment (allthesignals) [reply frontend/app/components/project-details.js]: Ah, it's like self-documentation. I didn't understand where `save` was coming from - realized it was being passed through.

Additionally, it's easier to unit test this component when it's clear what the main event of this component (this action) is a function.

Happy to remove but I think it's best to think of components as self-contained islands!

**Comment by pichot** [comment]
Looks good 👍

File: frontend/app/components/project-details.js (hunk 2 above)
Comment (pichot) [reply frontend/app/components/project-details.js]: Word.

**Pull request merged**